contracts/hot-cold-aa.sol

```diff
@@ -15,35 +15,28 @@ import "@matterlabs/zksync-contracts/l2/system-contracts/Constants.sol";
 // to call non-view function of system contracts
 import "@matterlabs/zksync-contracts/l2/system-contracts/libraries/SystemContractsCaller.sol";
 
-contract HotColdMultiSig is IAccount, IERC1271 {
+contract HotColdAA is IAccount, IERC1271 {
     // to get transaction hash
     using TransactionHelper for Transaction;
 
     // state variables for account owners
-    address public owner1;
-    address public owner2;
-
-    constructor(address _owner1, address _owner2) {
-        owner1 = _owner1;
-        owner2 = _owner2;
+    address public hotWallet;
+    address public coldWallet;
+    uint256 public maxValue;
+
+    constructor(address _hotWallet, address _coldWallet, uint256 _maxValue) {
+        hotWallet = _hotWallet;
+        coldWallet = _coldWallet;
+        maxValue = _maxValue;
     }
 
-    /*
-        Signature Validation Steps:
-
-        1. Check if the length of the received signature is correct
-        2. Extract the two signatures from the received multisig using the helper function
-            `extractEDCSASignature`
-        3. Check if the signatures are in the correct format using the helper function 
-            `checkValidEDCSASignatureFormat`
-        4. Extract the addresses from the transaction hash and each signature using the 
-            ECDSA.recover method
-        5. Check if the addresses extracted from the signatures match the owners of the account
-        6. Return the `EIP1271_SUCCESS_RETURN_VALUE` value on success or bytes4(0) if validation fails.
-     */
+    // TODO: check with Vlad on the usage of this one
+    modifier onlyAccount() {
```

Comment:
Usually, `msg.sender == address(this)` neened for the special auth. For instance to update critical account parameters one can make a transaction from account to itself and auth check would be `onlyAccount` modifier. 

It is some kind of protection to check whether account passsed valid signature and authorised the transaction, then allow account do a specific action by only checking that it called itself.